packages/x-date-pickers/src/dateViewRenderers/dateViewRenderers.tsx

```diff
@@ -16,13 +16,84 @@ export interface DateViewRendererProps<TDate, TView extends DateOrTimeView>
   focusedView: TView | null;
 }
 
-export const renderDateViewCalendar = <TDate extends unknown>(
-  props: DateViewRendererProps<TDate, any>,
-) => (
+export const renderDateViewCalendar = <TDate extends unknown>({
+  view,
+  onViewChange,
+  views,
+  focusedView,
+  onFocusedViewChange,
+  value,
+  defaultValue,
+  onChange,
+  className,
+  classes,
+  disableFuture,
+  disablePast,
+  minDate,
+  maxDate,
+  shouldDisableDate,
+  shouldDisableMonth,
+  shouldDisableYear,
+  reduceAnimations,
+  onMonthChange,
+  monthsPerRow,
+  onYearChange,
+  yearsPerRow,
+  defaultCalendarMonth,
+  components,
+  componentsProps,
+  slots,
+  slotProps,
+  loading,
+  renderLoading,
+  disableHighlightToday,
+  readOnly,
+  disabled,
+  showDaysOutsideCurrentMonth,
+  dayOfWeekFormatter,
+  sx,
+  autoFocus,
+  fixedWeekNumber,
+  displayWeekNumber,
+}: DateViewRendererProps<TDate, any>) => (
   <DateCalendar
-    {...props}
```

Comment:
What was the problem you are solving with the removal of spread? 🤔

Reply:
The renderer receives all the props passed to the pickers.
Including props only intended for the fields or for the picker layer (not for the view).
So if we forward everything and spread to the root, React complains that we are passing unwanted props to the root (things like `format` or `inputRef` will be passed to the DOM)